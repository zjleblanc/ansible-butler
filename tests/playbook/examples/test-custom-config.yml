- name: get network adapter information for each host
  hosts: all
  gather_facts: no

  tasks:
    - name: make absolutely sure the connection is active
      wait_for_connection:

    - name: get network connection name for private adapter
      win_shell: |
        foreach ($instance in (Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "Netenabled='True'")) {
            $instance_config = Get-CimInstance -ClassName WIn32_NetworkAdapterConfiguration -Filter "Index = '$($instance.Index)'"
            if ($instance_config.IPAddress -contains "{{ansible_host}}") {
                $instance.NetConnectionID
            }
        }
      changed_when: false
      register: network_connection_name

    - name: Create a resource group
      azure_rm_resourcegroup:
        name: "{{ resource_group }}"
        location: "{{ location }}"

    - name: Info about ec2 using ec2_instance_facts
      ec2_instance_facts:
        region: us-east-1
