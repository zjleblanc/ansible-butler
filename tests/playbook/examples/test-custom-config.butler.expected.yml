- name: get network adapter information for each host
  hosts: all
  gather_facts: no

  tasks:
    - name: make absolutely sure the connection is active
      ansible.builtin.wait_for_connection:

    - name: get network connection name for private adapter
      butler.windows.win_shell: |
        foreach ($instance in (Get-CimInstance -ClassName Win32_NetworkAdapter -Filter "Netenabled='True'")) {
            $instance_config = Get-CimInstance -ClassName WIn32_NetworkAdapterConfiguration -Filter "Index = '$($instance.Index)'"
            if ($instance_config.IPAddress -contains "{{ansible_host}}") {
                $instance.NetConnectionID
            }
        }
      changed_when: false
      register: network_connection_name

    - name: Create a resource group
      butler.azure.azure_rm_resourcegroup:
        name: "{{ resource_group }}"
        location: "{{ location }}"

    - name: Info about ec2 using ec2_instance_facts
      butler.aws.ec2_instance_facts:
        region: us-east-1
